sdk/core/azure-core/src/http/curl/curl_connection_pool_private.hpp

```diff
@@ -41,6 +45,10 @@ namespace Azure { namespace Core { namespace Http {
     // Give access to private to this tests class
     friend class Azure::Core::Test::CurlConnectionPool_connectionPoolTest_Test;
 #endif
+  private:
+    // The cttor and dttor of this member makes sure of calling the libcurl global init and cleanup
+    AZ_CORE_DLLEXPORT static CurlGlobalStateForAzureSdk CurlGlobalState;
```

Comment:
Can you guarantee that this is destroyed after the static state used by `CurlConnectionPool::CleanUp`?

Reply:
Fixing here: https://github.com/Azure/azure-sdk-for-cpp/pull/2105